### DISCUSSION TITLE
Faraway locations bloat save file sizes

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14613] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

The save file will keep track of every outpost, stores, events... By the time you reach the last biome, the save file size increases considerably _(which apparently causes problems for some people)_

**I'm unsure of the performance impact of this since save files aren't huge to begin with, BUT:**

If you are in biomes 3/4/5, it is very very unlikely that you'll ever go back to biomes 1/2 , so there's potential optimization by deleting unimportant data of those locations. 
You can potentially delete some data like Stores and Events on locations that are a certain amount of difficulty behind you (any location minus 30 difficulty?). 

Stores seem to be the biggest offenders in save file bloat (highlighted parts)
![image](https://github.com/user-attachments/assets/127ec48f-f3c1-423c-9548-aaad625fbcf7)

### Reproduction steps

Here's a save file from a campaign that reached hydrothermal wastes
[gamesession.zip](https://github.com/user-attachments/files/16988808/gamesession.zip)

### Bug prevalence

Just once

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
This has now been addressed in the unstable branch, and the changes will be included in the next release.
